diff --git a/Framework/PCAddFilesPanel.m b/Framework/PCAddFilesPanel.m
index 3851c19..0cbf559 100644
--- a/Framework/PCAddFilesPanel.m
+++ b/Framework/PCAddFilesPanel.m
@@ -107,9 +107,9 @@ static PCAddFilesPanel *addFilesPanel = nil;
 {
   NSString  *category = [fileTypePopup titleOfSelectedItem];
 
-  if ([_delegate respondsToSelector:@selector(categoryChangedTo:)])
+  if ([[self delegate] respondsToSelector:@selector(categoryChangedTo:)])
     {
-      [_delegate categoryChangedTo:category];
+      [[self delegate] categoryChangedTo:category];
     }
 }
 
diff --git a/Framework/PCButton.m b/Framework/PCButton.m
index 868defa..6ddb132 100644
--- a/Framework/PCButton.m
+++ b/Framework/PCButton.m
@@ -86,7 +86,7 @@
 {
   // If retain count is number of tracking rects + superview retain,
   // remove tracking rects. It seems that tracking rects retain this object.
-  if (_hasTooltips && [self retainCount] == [_tracking_rects count] + 1)
+  if (_hasTooltips && [self retainCount] == [[self _trackingRects] count] + 1)
     {
       [self removeAllToolTips];
     }
@@ -110,11 +110,12 @@
     {
       return;
     }
-    
-  j = [_tracking_rects count];
+  
+  NSArray *trackingRects = [self _trackingRects];
+  j = [trackingRects count];
   for (i = 0; i < j; i++)
     {
-      tr = [_tracking_rects objectAtIndex:i];
+      tr = [trackingRects objectAtIndex:i];
       
       // Save tooltip string for next addTrackingRect::::
       string = [(NSString *)tr->user_data copy];
diff --git a/Framework/PCProjectBuilder.m b/Framework/PCProjectBuilder.m
index df8b7d8..59d0358 100644
--- a/Framework/PCProjectBuilder.m
+++ b/Framework/PCProjectBuilder.m
@@ -873,7 +873,7 @@
   NSInteger position = 0;
   NSRange   range = NSMakeRange(position,1);
 
-  while ([[lineString substringFromRange:range] isEqualToString:@" "])
+  while ([[lineString substringWithRange:range] isEqualToString:@" "])
     {
       range.location = ++position;
     }
diff --git a/GNUmakefile.preamble b/GNUmakefile.preamble
index 0e315dc..d764e85 100644
--- a/GNUmakefile.preamble
+++ b/GNUmakefile.preamble
@@ -41,11 +41,17 @@ ADDITIONAL_INCLUDE_DIRS += -I./ -I./Headers
 # Additional LDFLAGS to pass to the linker
 ADDITIONAL_LDFLAGS += 
 
-# Additional library directories the linker should search
-ADDITIONAL_LIB_DIRS += -L./Framework/ProjectCenter.framework/$(GNUSTEP_TARGET_LDIR)
-
-# Additional LDFLAGS to pass to the linker
-ADDITIONAL_GUI_LIBS += -lProjectCenter
+ifeq ($(findstring darwin, $(GNUSTEP_TARGET_OS)), darwin)
+    # Additional library directories tihe linker should search
+    ADDITIONAL_LIB_DIRS += -F./Framework/
+    # Additional LDFLAGS to pass to the linker
+    ADDITIONAL_GUI_LIBS += -framework ProjectCenter
+else
+    # Additional library directories tihe linker should search
+    ADDITIONAL_LIB_DIRS += -L./Framework/ProjectCenter.framework/$(GNUSTEP_TARGET_LDIR)
+    # Additional LDFLAGS to pass to the linker
+    ADDITIONAL_GUI_LIBS += -lProjectCenter
+endif
 
 #
 # Flags dealing with installing and uninstalling
diff --git a/Modules/Debuggers/ProjectCenter/PTYView.m b/Modules/Debuggers/ProjectCenter/PTYView.m
index ec9da92..a4b01b7 100644
--- a/Modules/Debuggers/ProjectCenter/PTYView.m
+++ b/Modules/Debuggers/ProjectCenter/PTYView.m
@@ -61,9 +61,12 @@
 #endif
 
 
-#if !(defined (__NetBSD__)) && !(defined (__SOLARIS__)) && !(defined (__OpenBSD__)) && !(defined(__FreeBSD__))
+#if !(defined (__NetBSD__)) && !(defined (__SOLARIS__)) && !(defined (__OpenBSD__)) && !(defined(__FreeBSD__)) && !(defined(__APPLE__))
 #  include <pty.h>
 #endif
+#if defined (__APPLE__)
+#  include <util.h>
+#endif
 
 #import "PTYView.h"
 
diff --git a/Modules/GNUmakefile.bundles b/Modules/GNUmakefile.bundles
index 6da79f5..d864cee 100644
--- a/Modules/GNUmakefile.bundles
+++ b/Modules/GNUmakefile.bundles
@@ -17,8 +17,11 @@ endif
 ifeq ($(GNUSTEP_TARGET_OS), cygwin)
     DO_LIB_LINK=yes
 endif
+ifeq ($(findstring darwin, $(GNUSTEP_TARGET_OS)), darwin)
+    $(GNUSTEP_INSTANCE)_LIB_DIRS += -F../../../Framework/
+    $(GNUSTEP_INSTANCE)_BUNDLE_LIBS += -framework ProjectCenter
+endif
 ifeq ($(DO_LIB_LINK), yes)
     $(GNUSTEP_INSTANCE)_LIB_DIRS += -L../../../Framework/ProjectCenter.framework/$(GNUSTEP_TARGET_LDIR)
     $(GNUSTEP_INSTANCE)_BUNDLE_LIBS += -lProjectCenter
 endif
-
